File: typescript-connector/src/models/Certfication.ts
// Class from DFC Business Ontology: #Certfication
import { SemanticObject } from "../core/SemanticObject.js";
import { WhatSubject, type WhatSubjectParams } from "./WhatSubject.js";

export interface CertficationParams extends WhatSubjectParams {
  certiferReference?: string;
  certificationScore?: string;
  operatorId?: string;
  certifies?: string[];
}
export class Certfication extends WhatSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Certfication";
  }

  certiferReference?: string;
  certificationScore?: string;
  operatorId?: string;
  certifies?: string[];

  constructor(
    semanticId: string,
    params?: CertficationParams,
  ) {
    super(semanticId, params);
    this.certiferReference = params?.certiferReference;
    this.certificationScore = params?.certificationScore;
    this.operatorId = params?.operatorId;
    this.certifies = params?.certifies;
    this.semanticType = Certfication.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:certiferReference", () => this.certiferReference);
    this.registerSemanticProperty("dfc-b:certificationScore", () => this.certificationScore);
    this.registerSemanticProperty("dfc-b:operatorId", () => this.operatorId);
    this.registerSemanticProperty("dfc-b:certifies", () => this.certifies);
  }
  static {
    SemanticObject.typeRegistry.set(Certfication.SEMANTIC_TYPE, Certfication);
  }
}

File: ruby-gem/lib/models/certfication.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Certfication
require_relative '../core/semantic_object'
require_relative 'what_subject'


module DfcLinkmlConnector
  module Models
    class Certfication < WhatSubject
      SEMANTIC_TYPE = "dfc-b:Certfication".freeze

      # @return [String]
      attr_accessor :certifer_reference

      # @return [String]
      attr_accessor :certification_score

      # @return [String]
      attr_accessor :operator_id

      # @return [String]
      attr_accessor :certifies

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, certiferReference: nil, certificationScore: nil, operatorId: nil, certifies: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, certiferReference: nil, certificationScore: nil, operatorId: nil, certifies: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @certifer_reference = certiferReference
        @certification_score = certificationScore
        @operator_id = operatorId
        @certifies = certifies
        self.semanticType = "dfc-b:Certfication"
        registerSemanticProperty("dfc-b:certiferReference", &method("certifer_reference")).valueSetter = method("certifer_reference=")
        registerSemanticProperty("dfc-b:certificationScore", &method("certification_score")).valueSetter = method("certification_score=")
        registerSemanticProperty("dfc-b:operatorId", &method("operator_id")).valueSetter = method("operator_id=")
        registerSemanticProperty("dfc-b:certifies", &method("certifies")).valueSetter = method("certifies=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
